docs/utilities/idempotency.md

```diff
@@ -215,6 +217,70 @@ Resources:
     see 1WCU and 1RCU. Review the [DynamoDB pricing documentation](https://aws.amazon.com/dynamodb/pricing/) to
     estimate the cost.
 
+#### Using Redis as persistent storage layer
+
+##### Redis resources
+
+You need an existing Redis service before setting up Redis as persistent storage layer provider. You can also use Redis compatible services like [Amazon ElastiCache for Redis](https://aws.amazon.com/elasticache/redis/) as persistent storage layer provider.
+!!! tip "Tip:No existing Redis service?"
+If you don't have an existing Redis service, we recommend using DynamoDB as persistent storage layer provider.
+
+If you are using Redis you'll need to provide the Redis host, port, user and password as AWS Lambda environment variables.
```

Comment:
I think this should be configurable; I don't think we should force people to put passwords in env vars. Long term it would be cool if we had [IAM integration](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/auth-iam.html) here, short term enough to be able to provide the password somehow, having retrieved it e.g. using the `SecretsManagerProvider` ?

(apologies if you can already do this - i haven't got to the code yet!)

Reply:
For IAM Integration, I am not sure, this would be an extra possibility, since Amazon ElasticCache is the only possible Redis deployment that people might use.
For SecretsManagerProvider, sure, I agree. I wanted to do that too. Since this is already a lot of implementation details in this PR, I felt I should put it for review, to discuss the approach, list findings and decide which of them are important to be included with this PR and which can be a in a next step.

Reply:
I agree, and don't think either of these things needs to happen here. The whole redis rabbit whole we started down for a separate impl to prove the abstraction out; we have that without this other bits!